adds documentation for flag --primary-key-type=uuid passed to Rails model generator

jasonfb opened the pull request:
please see https://github.com/rails/rails/issues/46229

jasonfb committed:
adds documentation for flag --primary-key-type=uuid passed to Rails model generator

jasonfb commented on guides/source/active_record_postgresql.md:
```diff
@@ -504,6 +504,16 @@ irb> device.id
+(in this arbitrary example, a 'case' belongs to a device and so has a foriegn key `device_id` to the `devices` table we made above)
+
+`rails generate model Case device_id:uuid`
+
+
+
```
```suggestion
```

jasonfb commented on guides/source/active_record_postgresql.md:
```diff
@@ -504,6 +504,11 @@ irb> device.id
 
+To use the Rails model generator to make a table with UUID as the primary key, pass `--primary-key-type=uuid` as a flag to the model generator.
+
+When building a foreign key that will reference this a UUID, treat `uuid` as the field type, for example:
```
```suggestion
When building a foreign key that will reference this UUID, treat `uuid` as the native field type, for example:
```

eileencodes commented on guides/source/active_record_postgresql.md:
```diff
@@ -504,6 +504,11 @@ irb> device.id
 NOTE: `gen_random_uuid()` (from `pgcrypto`) is assumed if no `:default` option was
 passed to `create_table`.
 
+To use the Rails model generator to make a table with UUID as the primary key, pass `--primary-key-type=uuid` as a flag to the model generator.
```
```suggestion
To use the Rails model generator for a table using UUID as the primary key, pass `--primary-key-type=uuid` to the model generator.
```

eileencodes commented on guides/source/active_record_postgresql.md:
```diff
@@ -504,6 +504,11 @@ irb> device.id
 
+To use the Rails model generator to make a table with UUID as the primary key, pass `--primary-key-type=uuid` as a flag to the model generator.
+
+When building a foreign key that will reference this UUID, treat `uuid` as the native field type, for example:
```
```suggestion
When building a model with a foreign key that will reference this UUID, treat `uuid` as the native field type, for example:
```

eileencodes commented on guides/source/active_record_postgresql.md:
```diff
@@ -504,6 +504,11 @@ irb> device.id
+
+When building a foreign key that will reference this UUID, treat `uuid` as the native field type, for example:
+
+`rails generate model Case device_id:uuid`
```
This needs a new line but also needs to be code-blocked with ruby like the above examples.

```suggestion
`rails generate model Case device_id:uuid`

```

eileencodes reviewed (changes requested):
Hey @jasonfb thanks for the PR. This needs some changes before it's mergable.

1) Docs should be wrapped to 80 characters
2) New line code examples should be wrapped in a "```ruby" block, see above code for an example.
3) I've added some wording changes to make the language clearer.
4) When you're done squash your commits into 1 commit, we don't merge PRs with multiple commits unless they tell a specific and clear story.

Thanks!

jasonfb commented:
ok thanks @eileencodes for the feedback. I squashed it down. Not sure what you mean by "Docs should be wrapped to 80 characters"? I think all the lines I added are under 80 characters long. thanks.

eileencodes commented on guides/source/active_record_postgresql.md:
```diff
@@ -504,6 +504,21 @@ irb> device.id
 NOTE: `gen_random_uuid()` (from `pgcrypto`) is assumed if no `:default` option was
 passed to `create_table`.
 
+To use the Rails model generator for a table using UUID as the primary key, pass `--primary-key-type=uuid` to the model generator.
```
```suggestion
To use the Rails model generator for a table using UUID as the primary key, pass
`--primary-key-type=uuid` to the model generator.
```

eileencodes commented on guides/source/active_record_postgresql.md:
````diff
@@ -504,6 +504,21 @@ irb> device.id
+rails generate model Device --primary-key-type=uuid kind:string
+```
+
+When building a model with a foreign key that will reference this UUID, treat `uuid` as the native field type, for example:
````
```suggestion
When building a model with a foreign key that will reference this UUID, treat `uuid`
as the native field type, for example:
```

eileencodes reviewed (changes requested):
Wrapped to 80 characters means that lines shouldn't exceed 80 characters in length. I've left a code suggestion so you can see where the breaks should be 👍🏼

jasonfb commented:
oh, I got it. thank you. this is hard-wrapped to 80 chars now.

eileencodes commented:
Thank you @jasonfb!

jasonfb commented:
@eileencodes - you know I discovered a bug that makes this wrong — it appears to create a table that **_does_** have a ` id: :uuid ` set in the create_table definition, but then the generated migration doesn't actually contain the fields specified. 

so I guess the bug is that you can't use the flag and also pass fields to the model generator?

skipkayhil commented:
> so I guess the bug is that you can't use the flag and also pass fields to the model generator?

If you have reproduction steps, can you open a new issue? Replying to old PRs usually won't get as much visibility
